docs/developer/payments/price-calculations.mdx

```diff
@@ -0,0 +1,185 @@
+  1. recalculate amounts based on events without PSP reference
+  2. recalculate amounts based on `AUTHORIZATION` events
+  3. recalculate amounts based on `CHARGE` events
+  4. recalculate amounts based on `REFUND` events
+  5. recalculate amounts based on `CANCEL` events
+At each step, the amounts are incrementally adjusted by adding or subtracting values
+calculated in the current step to the totals from the previous step.
+
+### Assumptions
+
+#### 1. Event Grouping:
+  - Events are grouped based on their PSP reference and type of action (e.g., authorization, charge).
+  - Grouping ensures that events of the same type and PSP reference are properly matched to calculate amounts accurately.
+#### 2. Pending Amounts:
+  - Pending amounts (`authorizePendingAmount`, `chargePendingAmount`, `refundPendingAmount`, `cancelPendingAmount`) are increased only if a `REQUEST` event exists for the corresponding PSP reference.
+  - If a success or failure event is also associated with the same PSP reference, the system assumes the requested amount has already been processed, and the pending amount will not be increased.
+#### 3. `SUCCESS` and `FAILURE` Events:
+  - The transaction amount (e.g., `authorizedAmount`, `chargedAmount`) is increased only when a `SUCCESS` event exists.
+  - If both a `SUCCESS` and `FAILURE` event are present for the same PSP reference, the system compares their creation timestamps:
+    - If the `FAILURE` event is newer, the `SUCCESS` event is ignored.
+    - If the `SUCCESS` event is newer, it is used for the calculations.
+#### 4. Events Without PSP References:
+  - Not all events require a PSP reference, but they are still included in the transaction amount calculation. The following events do not need a PSP reference:
+    - `AUTHORIZATION_SUCCESS`
+    - `AUTHORIZATION_ADJUSTMENT`
+    - `CHARGE_SUCCESS`
+    - `CHARGE_BACK`
+    - `REFUND_SUCCESS`
+    - `REFUND_REVERSE`
+    - `CANCEL_SUCCESS`
+#### 5. Adjustment Events:
+  - The `AUTHORIZATION_ADJUSTMENT`, overrides previous amount from previous `AUTHORIZATION` events, this means that if an adjustment is present,
+  older authorization events are ignored, and the adjusted amount becomes the new authorized value.
+
+This logic ensures that transaction amounts reflect the current state of the transaction based on its event history while accounting for both processed and pending actions.
+
+### Calculation details
+:::note
+Events like `X_REQUEST`, generated by Saleor when calling the payment app, are excluded from calculations until a corresponding event with a valid `psp_reference` is received from the payment app.
+For example, an `AUTHORIZATION_REQUEST` event will not affect the calculations until an `AUTHORIZATION_SUCCESS` or a similar event containing a `psp_reference` is returned by the payment app.
+:::
+
+#### Authorized Value ([`transactionItem.authorizedAmount`](/api-reference/payments/objects/transaction-item#transactionitemauthorizedamountmoney---))
```

Comment:
Maybe worth mentioning that the below calculation flows are for the events with the same psp-refence?
For example:
```
- **GIVEN** there is a `CHARGE_SUCCESS` event
     - **WHEN** there is no `CHARGE_FAILURE`
```
We can have `CHARGE_SUCCESS` with PSP:123, and `CHARGE_FAILURE` with PSP:ABC. In that case FAILURE will not have any impact on SUCCESS, as PSP is different